simpleParser.go

```diff
@@ -547,7 +609,100 @@ func checkTokenFieldToken(token string) (string, ParseTokenType, error) {
 		err = fmt.Errorf("Invalid field: %v - cannot start or end with a period", token)
 	}
 
-	return token, TokenTypeField, err
+	// Field name must start with a backtick
+	validBegin := regexp.MustCompile("^`")
+	if !validBegin.MatchString(token) {
+		err = fmt.Errorf("Invalid field: %v - must start with a backtick", token)
+	}
+
+	// Field name must end with a backtick OR a closed bracket
+	validEnd := regexp.MustCompile("(`$)|(]$)")
+	if !validEnd.MatchString(token) {
+		err = ErrorMissingBacktickBracket
+	}
+
+	if err != nil {
+		return token, TokenTypeField, err
+	}
+
+	ctx.lastFieldTokens = make([]string, 0)
+	checkHelper := NewCheckFieldTokenHelper(token, &ctx.lastFieldTokens)
```

Comment:
if checkHelper is used only once, why do we need this object and the associated memory overhead? what benefit does it provide compared to a static method?

Reply:
That's true. Will change it.